### PR TITLE
Fix handling of empty messages

### PR DESCRIPTION
**Pull request by fooker**
This tries to solve #3.

### CODE REVIEW

File: bin/test.c [hunk 1]
```diff
@@ -237,6 +240,6 @@ void test_serialiser() {
 
 int main(int argc, char** argv) {
   test_parser();
-  test_serialiser();
+//  test_serialiser();
```
Comment (slaff) [inline bin/test.c]: Revert the line above.

File: src/parser.c [hunk 2]
```diff
@@ -487,6 +487,10 @@ static mqtt_parser_rc_t mqtt_parser_process(mqtt_parser_t* parser, mqtt_message_
         message->publish.content.length = parser->needs;
         message->publish.content.data   = NULL;
 
+        if(!parser->needs) {
+          goto DONE;
+	}
```
Comment (slaff) [inline src/parser.c]: Improve the indentation.

Comment (slaff) [reply src/parser.c]: You can run `make cs` to do the fixing for you. But this will require clang-format v5 or v6 to be installed on your system.

**Review by slaff — CHANGES_REQUESTED** [review]

File: src/parser.c (hunk 2 above)
Comment (fooker) [reply src/parser.c]: Done.

File: bin/test.c (hunk 1 above)
Comment (fooker) [reply bin/test.c]: Done.

File: bin/test.c [hunk 3]
```diff
@@ -130,6 +130,9 @@ void test_parser() {
       0x30, 0x17, 0x00, 0x0b, 0x53, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x54, 0x6f, 0x70, 0x69, 0x63, 0x48, 0x65, 0x6c, 0x6c,
       0x6f, 0x20, 0x4d, 0x51, 0x54, 0x54,
 
+      /* Publish without payload */
+      0x30, 0x0d, 0x00, 0x0b, 'S', 'a', 'm', 'p', 'l', 'e', 'T', 'o', 'p', 'i', 'c',
```
Comment (slaff) [inline bin/test.c:134]: One last thing -  I understand that this is just a sample of MQTT message without payload but it is a valid one? Is the MQTT protocol allowing publishing to a topic an empty message? If not - please, replace the example with a valid MQTT message that does not have payload.

Comment (fooker) [reply bin/test.c:134]: Yes, this one is valid. See http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718040
> The length of the payload can be calculated by subtracting the length of the variable header from the Remaining Length field that is in the Fixed Header. It is valid for a PUBLISH Packet to contain a zero length payload.

Comment (slaff) [reply bin/test.c:134]: Perfect, thanks.

**Review by slaff — APPROVED** [review]

**Pull request merged**